src/plugins/intel_npu/src/plugin/npuw/lazy_tensor.cpp

```diff
@@ -42,17 +57,52 @@ struct Const {
                 m_cached_ptr == other.m_cached_ptr);
     }
     ov::Tensor eval() const {
-        NPUW_ASSERT(m_node && "Const::eval() can only happen before detach");
-        return ov::npuw::util::tensor_from_const(m_node);
+        if (m_node) {
+            NPUW_ASSERT(m_node && "Const::eval() can only happen before detach");
+            return ov::npuw::util::tensor_from_const(m_node);
+        }
+
+        NPUW_ASSERT(m_read_from_weights && "Underlying data should have been read first!");
+        return m_read_from_weights;
+    }
+    void read_weight(const std::shared_ptr<ov::SharedBuffer<std::shared_ptr<ov::MappedMemory>>>& weights) {
+        NPUW_ASSERT(!m_node &&
+                    "LazyTensor can only read weight when it's being deserialized and not created from a Constant!");
+        m_read_from_weights = ov::Tensor(m_cached_type, m_cached_shape);
+        // Note: assumed to be called sequentially
+        std::memcpy(m_read_from_weights.data(), weights->get_ptr(m_offset), m_byte_size);
     }
     void detach() {
         m_node.reset();
+        if (m_read_from_weights) {
+            m_read_from_weights = ov::Tensor();
+        }
```

Comment:
I think it can be just down without an if.

Reply:
Done